packages/firestore/src/util/async_queue.ts

```diff
@@ -39,25 +58,37 @@ export class AsyncQueue {
    * Adds a new operation to the queue. Returns a promise that will be resolved
    * when the promise returned by the new operation is (with its value).
    *
-   * Can optionally specify a delay to wait before queuing the operation.
+   * Can optionally specify a delay (in milliseconds) to wait before queuing the
+   * operation.
    */
   schedule<T>(op: () => Promise<T>, delay?: number): Promise<T> {
     if (this.failure) {
       fail('AsyncQueue is already failed: ' + this.failure.message);
     }
 
     if ((delay || 0) > 0) {
-      this.delayedOpCount++;
-      const deferred = new Deferred<T>();
-      setTimeout(() => {
+      this.delayedOperationsCount++;
+      const nextIndex = this.delayedOperations.length;
```

Comment:
I think you can move this down and do:

const delayedOpIndex = this.delayedOperations.push(delayedOp);

Reply:
I don't actually need the index anymore and removed it altogether.